pandas/core/frame.py

```diff
@@ -4045,7 +4044,7 @@ def _sanitize_column(self, value) -> ArrayLike:
 
             # possibly infer to datetimelike
             if is_object_dtype(value.dtype):
-                value = maybe_infer_to_datetimelike(value)
+                value = Series(value)._values
```

Comment:
this makes this heavier weight. do we want to do this?

Reply:
sanitize_array should be equivalent